Cython/Compiler/PyrexTypes.py

```diff
@@ -2860,6 +2865,22 @@ def assignable_from_resolved_type(self, other_type):
             return self.base_type.is_void or self.base_type.same_as(other_type.base_type)
         return 0
 
+    def assignment_failure_extra_info(self, src_type):
+        if self.base_type.is_cfunction and src_type.is_ptr:
+            src_type = src_type.base_type.resolve()
+        if self.base_type.is_cfunction and src_type.is_cfunction:
+            copied_src_type = copy.copy(src_type)
+            # make the exception values the same as us
+            copied_src_type.exception_check = self.base_type.exception_check
+            copied_src_type.exception_value = self.base_type.exception_value
+            if self.base_type.pointer_assignable_from_resolved_type(copied_src_type):
+                # the only reason we can't assign is because of exception incompatibility
+                msg = "Exception specifications are incompatible."
```

Comment:
I'm not sure "exception specification" is something we use in the documentation, so it might be difficult to look up. I think we usually talk about "exception values", maybe "exception return values", although that's more constrained than "specification".

Reply:
We use it a few times in the docs (and at least of of those times was written by someone who wasn't me)

https://cython.readthedocs.io/en/latest/src/userguide/language_basics.html#language-basics
> cdef functions that are not extern are implicitly declared with a suitable exception specification for the return type

https://cython.readthedocs.io/en/latest/src/userguide/migrating_to_cy30.html
> There is an easy-to-encounter performance pitfall here with nogil functions with an implicit exception specification of except *.

https://cython.readthedocs.io/en/latest/src/userguide/nogil.html
> Be aware that a function with an except * exception specification (typically functions returning void) will be expensive to call because Cython will need to temporarily reacquire the GIL after every call to check the exception state. Most other exception specifications are cheap to handle in a nogil block since the GIL is only acquired if an exception is actually thrown.

> In most cases this is efficient since Cython is able to use the function’s exception specification to check for an error

So I'm not feeling too bad about it.

However; I'll replace it with "exception values"